clients/unshaded/src/main/java/tachyon/client/RemoteBlockInStream.java

```diff
@@ -103,6 +104,9 @@
    */
   private static final int MAX_REMOTE_READ_ATTEMPTS = 2;
 
+  /** A reference to the current reader so we can clear it after reading is finished. */
+  private static RemoteBlockReader sUnclearedReader = null;
```

Comment:
I don't think this should be 'static'.

Also, maybe naming it as "mCurrentReader" would make more sense.